File: TDDMicroExercises.ProposedSolution/Javascript/turn-ticket-dispenser/ticket-dispenser.js
TicketDispenser = function (turnNumberSequence) {
	this._turnNumberSequence = turnNumberSequence;
};

TicketDispenser.Create = function () {
	var turnNumberSequence = new TurnNumberSequence();
	return new TicketDispenser(turnNumberSequence);
};

TicketDispenser.prototype = {
	
	getTurnTicket: function () {

		var newTurnNumber = this._turnNumberSequence.getNextTurnNumber();
		var newTurnTicket = new TurnTicket(newTurnNumber);

		return newTurnTicket;
	}

};

File: Swift/Sources/TurnTicket/TicketDispenser.swift
import Foundation

public class TicketDispenser {
    public func getTurnTicket() -> TurnTicket {
        let newTurnNumber = TurnNumberSequence.getNextTurnNumber()
        let newTurnTicket = TurnTicket(_turnNumber: newTurnNumber)

        return newTurnTicket
    }
}
